### PR TITLE
Add support for bond interface in net-traffic

### PR DESCRIPTION
**Pull request by jeffothy**
Use case:
I use a bond as my main network interface, this breaks with the [most recent change](https://github.com/regolith-linux/regolith-i3xrocks-config/commit/0b12b5a93a250992f504aea2dd0dc9bdc97f0ff4) in net-traffic to use `/proc/net/route` for finding the interface with statistics to be used on the net-traffic i3xrocks script.

Changes:
1. Add a mocked /proc/net/route to the net traffic test
2. Allow interfaces with default routes starting with `bond`, use the `active_slave` of the bond to detect traffic for the bond interface
3. Add a new test to validate bond net traffic.

Note: this would be my first contribution to the project, so I'm definitely more than happy to take input/feedback here!

### CODE REVIEW

**Comment by kgilmer** [comment]
@winks , any concerns here?

**Comment by winks** [comment]
Looks awesome, sorry for breaking your use case, I hadn't thought of bond interfaces. It makes me think again how we could avoid the downsides of the old "filter out" approach and also the current "filter in" :/

File: scripts/net-traffic [hunk 1]
```diff
@@ -17,10 +17,16 @@ NUMFMT="numfmt --to iec --format %f --padding 5"
 
 # determine the net interface name, but only use one interface
 # there could be several default routes, if you have a VPN for example
-IF="${BLOCK_INSTANCE:-$(awk '$2 == 00000000 && $1 ~ /^(wl|en|eth)/ { print $1; exit }' < /proc/net/route)}"
+IF="${BLOCK_INSTANCE:-$(awk '$2 == 00000000 && $1 ~ /^(wl|en|eth|bond)/ { print $1; exit }' < /proc/net/route)}"
```
[the remark this answers is not in the document]
Comment (winks) [reply scripts/net-traffic:20]: I guess we should use the most likely configuration for most people (but what is that?), the important change was "only one line and not several", but maybe the exclude list is better, esp. regarding the wireguard remark I left in my PR iirc.

[the remark this answers is not in the document]
Comment (kgilmer) [reply scripts/net-traffic:20]: Is this a case where we are unsure of what a given user's environment may contain?  Would using an Xresource variable to specify network devices (or something else) be a way of moving forward?  Something else?

[the remark this answers is not in the document]
Comment (winks) [reply scripts/net-traffic:20]: maybe an overridable default would make the most sense, yeah.

I don't have a real argument to convert it to an include list (and as we saw there was a problem) and it worked before, so I'd suggest rolling back to the exclude list but making it overridable would be the best of both worlds.

@jeffothy does that work for you?

[the remark this answers is not in the document]
Comment (jeffothy) [reply scripts/net-traffic:20]: @winks Yes, I think rolling back to the exclude list is the right thing to do. I think my code suggestion in the first comment in this thread is still my preference (as opposed to a pure revert). It removes the need to add `|bond` in this case.

I'd say making it overiddable should be deferred, or at the very least left for a subsequent PR.

[the remark this answers is not in the document]
Comment (jeffothy) [reply scripts/net-traffic]: I'll resolve this conversation now. And if there's nothing else then we should be good for approval :+1:

**Review by kgilmer — APPROVED** [review]
LGTM

**Comment by kgilmer** [comment]
@jeffothy thanks for your PR and @winks thanks for helping out!

**Pull request merged**